drivers/modem/quectel-bg9x.c

```diff
@@ -953,6 +953,64 @@ static const struct setup_cmd setup_cmds[] = {
 	SETUP_CMD_NOHANDLE("AT+QICSGP=1,1,\"" MDM_APN "\",\"" MDM_USERNAME "\", \"" MDM_PASSWORD "\",1"),
 };
 
+
+/* Func: modem_pdp_context_open
+ * Desc: This helper function is called from modem_setup, and is
+ * used to open the PDP context. If there is trouble opening the
+ *  PDP context, we try to deactive and reactive MDM_PDP_ACT_RETRY_COUNT times.
+ *  If it fails, we return an error.
+ */
+
+
+static int modem_pdp_context_open(const int contextID)
+{
+
+	int ret;
+	int pdp_act_retry_count = 0;
+	char buf_act[sizeof("AT+QIACT=##")] = {0};
+	char buf_deact[sizeof("AT+QIDEACT=##")] = {0};
+
+	if (contextID < MDM_MIN_PDP_CONTEXT_ID || contextID > MDM_MAX_PDP_CONTEXT_ID) {
+		LOG_ERR("WRONG CONTEXT ID USED : %d ; SHOULD BE BETWEEN %d AND %d",
+				contextID, MDM_MIN_PDP_CONTEXT_ID, MDM_MAX_PDP_CONTEXT_ID);
+				return -1;
+	}
+
+	snprintk(buf_act, sizeof(buf_act), "AT+QIACT=%d", contextID);
+	snprintk(buf_deact, sizeof(buf_deact), "AT+QIDEACT=%d", contextID);
+
+	ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, buf_act, &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+	/* If there is trouble opening the PDP context, we try to deactivate/reactive it. */
+	while (ret == -EIO && pdp_act_retry_count < MDM_PDP_ACT_RETRY_COUNT) {
+
+		ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, buf_deact, &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+		/* If there's any error for AT+QIDEACT, restart the module. */
+		if (ret != 0) {
+			return ret;
+		}
+
+		ret = modem_cmd_send(&mctx.iface, &mctx.cmd_handler,
+			     NULL, 0U, buf_act, &mdata.sem_response,
+			     MDM_CMD_TIMEOUT);
+
+		pdp_act_retry_count++;
+	}
+
+	if (ret != -EIO || pdp_act_retry_count >= MDM_PDP_ACT_RETRY_COUNT) {
+		LOG_ERR("RETRIED ACTIVATING/DEACTIVATING TOO MUCH TIMES");
+	}
```

Comment:
"RETRIED ACTIVATING/DEACTIVATING TOO MUCH TIMES" is truth when `pdp_act_retry_count >= MDM_PDP_ACT_RETRY_COUNT` is true. If `ret != -EIO`, then this message is misleading.